Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -200,13 +254,23 @@ public void UnstickFromEntity(EntityUid uid, EntityUid user, StickyComponent? co
         }
 
         // show message to user
-        if (component.UnstickPopupSuccess != null)
+        if (component.UnstickPopupSuccess != null && user != null)
         {
-            var msg = Loc.GetString(component.UnstickPopupSuccess);
-            _popupSystem.PopupEntity(msg, user, user);
+            var msg = Loc.GetString(component.UnstickPopupSuccess, ("item", Name(uid)));
+            _popupSystem.PopupEntity(msg, user.Value, user.Value);
         }
 
         component.StuckTo = null;
         RaiseLocalEvent(uid, new EntityUnstuckEvent(target, user), true);
     }
+
+    public void TransferStickySurfaceFromDestroyedEntity(EntityUid uid, TransferStickySurfaceOnSpawnBehaviorComponent component, OnSpawnFromSpawnEntitiesBehaviourEvent ev)
+    {
+        if (EntityManager.TryGetComponent<StickyComponent>(uid, out var ownerStickyComp) &&
+            EntityManager.TryGetComponent<StickyComponent>(ev.Spawned, out var spawnedStickyComp) &&
```

Comment:
proxy methods, rename to On`EventName`, make it private and move it alongside the other event handlers (preferably following the order on the subscriptions)